bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/GhostSupport.java

```diff
@@ -0,0 +1,61 @@
+package edu.gemini.wdba.tcc;
+
+import edu.gemini.spModel.gemini.ghost.Ghost;
+
+import java.util.Objects;
+
+public class GhostSupport implements ITccInstrumentSupport {
+    private ObservationEnvironment oe;
+
+    private GhostSupport(final ObservationEnvironment oe) {
+        Objects.requireNonNull(oe, "Observation environment can not be null");
+        this.oe = oe;
+    }
+
+    /**
+     * Factor for creating a new GHOST Instrument Support.
+     *
+     * @param oe The current ObservationEnvironment
+     * @return A new instance
+     * @throws NullPointerException returned if the ObservationEnvironment is null.
+     */
+    static public ITccInstrumentSupport create(final ObservationEnvironment oe) throws NullPointerException {
```

Comment:
It's not often you see an unchecked exception in the signature.  I suppose the other `FooSupport` instances do the same thing.

Reply:
Yes, I thought this was weird, too, but I copied this from the other `...Support` classes (as clearly others did, since most instruments were labeled as TReCS in their comments 😆 .)